crates/world/pool/src/validator.rs

```diff
@@ -179,16 +182,25 @@ where
                         &valid_roots,
                         self.max_pbh_nonce.load(Ordering::Relaxed),
                     )?;
-
                     Ok::<PbhPayload, WorldChainPoolTransactionError>(payload)
                 })
                 .collect::<Result<Vec<PbhPayload>, WorldChainPoolTransactionError>>()
             {
-                Ok(payloads) => {
-                    aggregated_payloads.extend(payloads);
-                }
+                Ok(payloads) => payloads,
                 Err(err) => return err.to_outcome(tx),
+            };
+
+            // Now check for duplicate nullifier_hashes
```

Comment:
Nit: Since `seen_nullifier_hashes` is only used here can we initialize the hashset right above this rather than before parsing the PBH payloads?